From 1149fcf0f4c8889f7b42dc7df712bbbfc2def281 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 22 Nov 2019 08:06:52 -0600
Subject: [PATCH 15/34] ASoC: imx-wm8962: Use EPROBE_DEFER if codec platform
 isn't found

The imx-wm8962 driver requires the wm8962 codec to be loaded first.
Under some circumstances this doesn't happen, and the imx-wm8962 probe
fails to load and never tries again even after the codec does.

This patch changes the return value from EINVAL to EPROBE_DEFER
to probe again when the codec isn't loaded first.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 sound/soc/fsl/imx-wm8962.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-wm8962.c b/sound/soc/fsl/imx-wm8962.c
index 3bdac06f5563..06301e0cfcf8 100644
--- a/sound/soc/fsl/imx-wm8962.c
+++ b/sound/soc/fsl/imx-wm8962.c
@@ -655,7 +655,7 @@ static int imx_wm8962_probe(struct platform_device *pdev)
 	codec_dev = of_find_i2c_device_by_node(codec_np);
 	if (!codec_dev || !codec_dev->dev.driver) {
 		dev_err(&pdev->dev, "failed to find codec platform device\n");
-		ret = -EINVAL;
+		ret = -EPROBE_DEFER;
 		goto fail;
 	}
 
-- 
2.17.1

